diff --git a/BUILD b/BUILD
index 4745cc2dd5..42a1c94606 100644
--- a/BUILD
+++ b/BUILD
@@ -76,9 +76,7 @@ config_setting(

 config_setting(
     name = "android",
-    values = {"crosstool_top": "//external:android/crosstool"},
-    # TODO: Use constraint_values to detect android after Bazel 7.0 platforms migration is finished
-    # constraint_values = [ "@platforms//os:android" ],
+    constraint_values = [ "@platforms//os:android" ],
 )

 config_setting(
