diff --git a/bazel/rules.bzl b/bazel/rules.bzl
index 7be5b02613..9875ca8528 100644
--- a/bazel/rules.bzl
+++ b/bazel/rules.bzl
@@ -352,10 +352,17 @@ def perfetto_cc_tp_tables(name, srcs, outs, deps = [], **kwargs):

     cmd = ["$(location " + name + "_tool)"]
     cmd += ["--gen-dir", "$(RULEDIR)"]
-    cmd += ["--inputs", "$(SRCS)"]
+    # Do not use $(SRCS) expansion for --inputs.
+    # Arguments given to --inputs are converted into python modules, using
+    # brittle string manipulation. Instead, do the string manipulation here.
+    # If $(SRCS) expansion is used, filepaths can contain a path prefix, making
+    # it difficult to get the correct python module name.
+    # The py_library above, ${name}_lib, bundles python modules to be
+    # available on the python path for the py_binary, ${name}_tool.
+    module_names = [src.replace("/", ".").removesuffix(".py") for src in srcs]
+    cmd += ["--inputs", " ".join(module_names)]
     if PERFETTO_CONFIG.root != "//":
         cmd += ["--import-prefix", PERFETTO_CONFIG.root[2:]]
-        cmd += ["--relative-input-dir", PERFETTO_CONFIG.root[2:]]

     perfetto_genrule(
         name = name + "_gen",
diff --git a/tools/gen_tp_table_headers.py b/tools/gen_tp_table_headers.py
index c53602ede1..abbf89c4a1 100755
--- a/tools/gen_tp_table_headers.py
+++ b/tools/gen_tp_table_headers.py
@@ -69,13 +69,16 @@ def main():
   def get_relin_path_from_module_path(module_path: str):
     return module_path[module_path.rfind(os.sep + 'src') + 1:]

-  modules = [
-      # On Windows the path can contain '/' or os.sep, depending on how this
-      # script is executed. So we need to replace both.
-      os.path.splitext(
-          get_relin_path(i).replace('/', '.').replace(os.sep, '.'))[0]
-      for i in args.inputs
-  ]
+  def to_module_name(module_input: str):
+    module = get_relin_path(module_input)
+    if module.endswith('.py'):
+      module = module[:-3]
+    # On Windows the path can contain '/' or os.sep, depending on how this
+    # script is executed. So we need to replace both.
+    return module.replace('/', '.').replace(os.sep, '.')
+
+
+  modules = [to_module_name(i) for i in args.inputs]
   headers: Dict[str, Header] = {}
   for table in parse_tables_from_modules(modules):
     input_path = get_relin_path_from_module_path(table.table.python_module)
