diff --git a/go/private/rules/binary.bzl b/go/private/rules/binary.bzl
index f3f2f07c..a2e9257c 100644
--- a/go/private/rules/binary.bzl
+++ b/go/private/rules/binary.bzl
@@ -513,9 +513,10 @@ exit /b %GO_EXIT_CODE%
             content = cmd,
         )
         ctx.actions.run(
-            executable = bat,
+            executable = "cmd.exe",
+            arguments = ["/S", "/C", bat.path.replace("/", "\\")],
             inputs = depset(
-                ctx.files.srcs + [sdk.go],
+                ctx.files.srcs + [sdk.go, bat],
                 transitive = [sdk.headers, sdk.srcs, sdk.tools],
             ),
             outputs = [out, gotmp],
