diff --git a/rules/aar_import/impl.bzl b/rules/aar_import/impl.bzl
index 527c5a5..10b6b19 100644
--- a/rules/aar_import/impl.bzl
+++ b/rules/aar_import/impl.bzl
@@ -387,6 +387,13 @@ def _validate_rule(
         args.add("-has_lint_jar")
     args.add("-output", validation_output)

+    if checks.executable.extension == "bat":
+        # disable the validation check on Windows.
+        # This should not be necessary after updating to Bazel 8.
+        # See https://github.com/bazelbuild/bazel/issues/11636
+        ctx.actions.write(validation_output, "")
+        return validation_output
+
     ctx.actions.run(
         executable = checks,
         arguments = [args],
